pkg/controllers/machine/disruption/controller.go

```diff
@@ -108,6 +99,84 @@ func (c *Controller) Reconcile(ctx context.Context, machine *v1alpha5.Machine) (
 	return result.Min(results...), errs
 }
 
+type NodeClaimController struct {
+	*Controller
+}
+
+func NewNodeClaimController(clk clock.Clock, kubeClient client.Client, cluster *state.Cluster, cloudProvider cloudprovider.CloudProvider) corecontroller.Controller {
+	return corecontroller.Typed[*v1beta1.NodeClaim](kubeClient, &NodeClaimController{
+		Controller: NewController(clk, kubeClient, cluster, cloudProvider),
+	})
+}
+
+func (c *NodeClaimController) Name() string {
+	return "nodeclaim.disruption"
+}
+
+func (c *NodeClaimController) Reconcile(ctx context.Context, nodeClaim *v1beta1.NodeClaim) (reconcile.Result, error) {
+	return c.Controller.Reconcile(ctx, nodeClaim)
+}
+
+func (c *NodeClaimController) Builder(ctx context.Context, m manager.Manager) corecontroller.Builder {
+	return corecontroller.Adapt(controllerruntime.
+		NewControllerManagedBy(m).
+		For(&v1beta1.NodeClaim{}, builder.WithPredicates(
+			predicate.Or(
+				predicate.GenerationChangedPredicate{},
```

Comment:
?

Reply:
This is there to avoid re-queueing on the machine/nodeClaim disruption status condition patch. So, we will only re-queue when a machine/nodeClaim status condition that affects its readiness changes OR the spec changes (which could affect drift). This is more of a performance concern than anything and doesn't tangibly affect the logic of the code.

Reply:
Kinda wish this was the controllerruntime default. I should use it more :)